go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/openstack-k8s-operators/infra-operator
 
-go 1.19
+go 1.21
```

Comment:
lets make it consistent and either add go toolchain here or remove it from the apis/go.mod as well.

Reply:
added toolchain here